src/persist-client/src/internal/compact.rs

```diff
@@ -458,6 +430,218 @@ where
+                    && single_nonempty_batch.desc.since() != &Antichain::from_elem(T::minimum())
+                {
+                    metrics.compaction.fast_path_eligible.inc();
+                }
+            }
+
+            // Reserve space for the in-progress part to be held in-mem representation and columnar -
+            let in_progress_part_reserved_memory_bytes = 2 * cfg.batch.blob_target_size;
+            // - then remaining memory will go towards pulling down as many runs as we can.
+            // We'll always do at least two runs per chunk, which means we may go over this limit
+            // if parts are large or the limit is low... though we do at least increment a metric
+            // when that happens.
+            let run_reserved_memory_bytes = cfg
+                .compaction_memory_bound_bytes
+                .saturating_sub(in_progress_part_reserved_memory_bytes);
+
+            let ordered_runs =
+                Self::order_runs(&req, cfg.batch.preferred_order, &*blob, &*metrics).await?;
+            for (runs, run_chunk_max_memory_usage) in
+                Self::chunk_runs(&ordered_runs, &cfg, &*metrics, run_reserved_memory_bytes)
+            {
+                metrics.compaction.chunks_compacted.inc();
+                metrics
+                    .compaction
+                    .runs_compacted
+                    .inc_by(u64::cast_from(runs.len()));
+
+                // given the runs we actually have in our batch, we might have extra memory
+                // available. we reserved enough space to always have 1 in-progress part in
+                // flight, but if we have excess, we can use it to increase our write parallelism
+                let extra_outstanding_parts = (run_reserved_memory_bytes
+                    .saturating_sub(run_chunk_max_memory_usage))
+                    / cfg.batch.blob_target_size;
+                let mut run_cfg = cfg.clone();
+                run_cfg.batch.batch_builder_max_outstanding_parts = 1 + extra_outstanding_parts;
+                let batch = Self::compact_runs(
+                    &run_cfg,
+                    &req.shard_id,
+                    &req.desc,
+                    runs,
+                    Arc::clone(&blob),
+                    Arc::clone(&metrics),
+                    Arc::clone(&shard_metrics),
+                    Arc::clone(&isolated_runtime),
+                    write_schemas.clone(),
+                )
+                .await?;
+                let (parts, run_splits, run_meta, updates) =
+                    (batch.parts, batch.run_splits, batch.run_meta, batch.len);
+                assert!(
+                    (updates == 0 && parts.len() == 0) || (updates > 0 && parts.len() > 0),
+                    "updates={}, parts={}",
+                    updates,
+                    parts.len(),
+                );
+
+                if updates == 0 {
+                    continue;
+                }
```

Comment:
This is unnecessary / has no effect currently, right?

If so let's skip it... as soon as we're doing actual incremental compactions, we'll want to treat these zero-output runs the same as every other run.